### PR TITLE
Add choice of display to --image.

### PR DESCRIPTION
**Pull request by nuew**
Additionally
  - Made background colors display when an image is enabled if
a --color is specified.
  - Link CJson to swaylock.
  - Add the --socket option to swaylock.

Sorry about having to make a new pull request.

### CODE REVIEW

File: swaylock/main.c [hunk 1]
```diff
@@ -221,18 +243,22 @@ int main(int argc, char **argv) {
...
-		"  -s, --scaling              Scaling mode: stretch, fill, fit, center, tile.\n"
-		"  -t, --tiling               Same as --scaling=tile.\n"
-		"  -v, --version              Show the version number and quit.\n"
-		"  -i, --image <path>         Display the given image.\n";
+		"  -h, --help                      Show help message and quit.\n"
+		"  -c, --color <rrggbb[aa]>        Turn the screen into the given color instead of white.\n"
+		"  -s, --scaling                   Scaling mode: stretch, fill, fit, center, tile.\n"
+		"  -t, --tiling                    Same as --scaling=tile.\n"
+		"  -v, --version                   Show the version number and quit.\n"
+		"  -i, --image [<display>:]<path>  Display the given image.\n"
```
Comment (SirCmpwn) [inline swaylock/main.c]: The language is "output", rather than display.

**Comment by SirCmpwn** [comment]
What happens if I specify one image with an output anme, but don't specify enough images for all outputs?

**Comment by nuew** [comment]
The output is drawn completely transparent, unless --color is specified.

**Comment by SirCmpwn** [comment]
Alright, that seems fair to me.

File: swaylock/main.c [hunk 2]
```diff
@@ -259,14 +286,39 @@ int main(int argc, char **argv) {
...
 		case 'i':
-			image_path = optarg;
+		{
+			char *image_path = strchr(optarg, ':');
+			if (image_path == NULL) {
+				if (num_images == 0) {
+					images = load_image(optarg);
+					num_images = -1;
+				} else {
+					fprintf(stderr, "display must be defined for all --images or no --images.\n");
```
Comment (SirCmpwn) [inline swaylock/main.c]: s/display/output/

File: swaylock/main.c [hunk 3]
```diff
@@ -259,14 +286,39 @@ int main(int argc, char **argv) {
...
 		case 'i':
-			image_path = optarg;
+		{
+			char *image_path = strchr(optarg, ':');
+			if (image_path == NULL) {
+				if (num_images == 0) {
+					images = load_image(optarg);
+					num_images = -1;
+				} else {
+					fprintf(stderr, "output must be defined for all --images or no --images.\n");
```
Comment (mikkeloscar) [inline swaylock/main.c]: Use `sway_log(L_ERROR, "...");` instead of `fprintf`.

Comment (SirCmpwn) [inline swaylock/main.c]: :+1:

**Comment by nuew** [comment]
huh... apparently '`' is a comment in a git commit.

**Comment by SirCmpwn** [comment]
LGTM, will merge when build passes.

**Pull request merged**